Source/MLX/MLXArray.swift

```diff
@@ -255,53 +353,17 @@ public final class MLXArray {
-        self.eval()
-
-        var array_ctx = self.ctx
-        var free = false
         if type.dtype != self.dtype {
-            array_ctx = mlx_astype(self.ctx, type.dtype.cmlxDtype, StreamOrDevice.default.ctx)
-            mlx_array_eval(array_ctx)
-            free = true
+            return self.asType(type).asArray(type)
         }
 
-        // can't do it inside the else as it will free at the end of the block
-        defer { if free { mlx_free(array_ctx) } }
+        self.eval()
 
-        func convert(_ ptr: UnsafePointer<T>) -> [T] {
-            Array(UnsafeBufferPointer(start: ptr, count: self.size))
-        }
-
-        switch type {
-        case is Bool.Type: return convert(mlx_array_data_bool(array_ctx) as! UnsafePointer<T>)
-        case is UInt8.Type: return convert(mlx_array_data_uint8(array_ctx) as! UnsafePointer<T>)
-        case is UInt16.Type: return convert(mlx_array_data_uint16(array_ctx) as! UnsafePointer<T>)
-        case is UInt32.Type: return convert(mlx_array_data_uint32(array_ctx) as! UnsafePointer<T>)
-        case is UInt64.Type: return convert(mlx_array_data_uint64(array_ctx) as! UnsafePointer<T>)
-        case is Int8.Type: return convert(mlx_array_data_int8(array_ctx) as! UnsafePointer<T>)
-        case is Int16.Type: return convert(mlx_array_data_int16(array_ctx) as! UnsafePointer<T>)
-        case is Int32.Type: return convert(mlx_array_data_int32(array_ctx) as! UnsafePointer<T>)
-        case is Int64.Type: return convert(mlx_array_data_int64(array_ctx) as! UnsafePointer<T>)
-        case is Int.Type:
-            // Int and Int64 are the same bits but distinct types. coerce pointers as needed
-            let pointer = mlx_array_data_int64(array_ctx)
-            let bufferPointer = UnsafeBufferPointer(start: pointer, count: self.size)
-            return bufferPointer.withMemoryRebound(to: Int.self) { buffer in
-                Array(buffer) as! [T]
-            }
-        #if !arch(x86_64)
-            case is Float16.Type:
-                return convert(mlx_array_data_float16(array_ctx) as! UnsafePointer<T>)
-        #endif
-        case is Float32.Type: return convert(mlx_array_data_float32(array_ctx) as! UnsafePointer<T>)
-        case is Float.Type: return convert(mlx_array_data_float32(array_ctx) as! UnsafePointer<T>)
-        case is Complex<Float32>.Type:
-            let ptr = UnsafeBufferPointer(
-                start: UnsafePointer<Complex<Float32>>(mlx_array_data_complex64(ctx)),
-                count: self.size)
-            return Array(ptr) as! [T]
-        default:
-            fatalError("Unable to get item() as \(type)")
+        return [T](unsafeUninitializedCapacity: self.size) { destination, initializedCount in
+            let source = UnsafeRawBufferPointer(
+                start: mlx_array_data_uint8(self.ctx), count: physicalSize * itemSize)
+            copy(from: source, to: UnsafeMutableRawBufferPointer(destination))
+            initializedCount = self.size
```

Comment:
Since I need to deal with bytes I don't need all the pointer type conversion -- just handle it as a byte pointer.